modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -1342,13 +1342,13 @@ class SettingsImpl @Inject constructor(
         setBoolean(NOTIFICATIONS_DISABLED_MESSAGE_SHOWN, value)
     }
 
-    override fun setLastSelectedSubscriptionTier(tier: Subscription.SubscriptionTier) {
-        setString(LAST_SELECTED_SUBSCRIPTION_TIER_KEY, tier.name)
+    override fun setLastSelectedSubscriptionTier(tier: SubscriptionTier) {
+        setString(LAST_SELECTED_SUBSCRIPTION_TIER_KEY, tier.label)
     }
 
     override fun getLastSelectedSubscriptionTier() =
         getString(LAST_SELECTED_SUBSCRIPTION_TIER_KEY)?.let {
-            Subscription.SubscriptionTier.valueOf(it)
+            SubscriptionTier.fromString(it)
```

Comment:
This is a safe change. The function will return `null` in the worst case and the code is used only to determine previously selected tier indication during onboarding. I think it's simpler to just do it this way than to provide a migration.